docs/en/administration/Query_planning.md

```diff
@@ -237,6 +237,26 @@ REFRESH ASYNC
 AS SELECT /*+ SET_VAR(query_timeout=500) */ * from dual;
 ~~~
 
+### User-defined variable hint
+
+You can set one or more [user-defined variables](../reference/user_defined_variables.md) by using the `SET_USER_VARIABLE` hint in the form of the syntax `/*+ SET_USER_VARIABLE(...) */` in SELECT statements, or in the SELECT clause.
+
+If a query references the result of a scalar subquery or scalar expression, you can use a user-defined variable hint to set that subquery or expression as a user-defined variable and then run the query. This not only can help avoid duplicate computation, but also does not influence the entire session because user-defined variable hints take effect at the statement level, unlike the [general usage of user-defined variables](../reference/user_defined_variables.md) that take effect at the session level.
```

Comment:
you can use a user-defined variable hint --- you can user the `SET_USER_VARIABLE` hint to 这里就是特指这个 hint，不能用 a

Reply:
This not only avoids duplicate computation, but also eliminates the influence on the entire session because the `SET_USER_VARIABLE` hint takes effect at the statement level, not the session level like user-defined variables.